dev/domain-controller/new-agent-vm/Install-PCoIPAgent.ps1

```diff
@@ -1,41 +1,246 @@
+                Write-Verbose "Starting to Install PCoIPAgent"
 
-                $source = "https://teradeploy.blob.core.windows.net/binaries/PCoIP_agent_release_installer_2.7.0.4060_standard.exe"
-                $dest = "C:\WindowsAzure\PCoIPAgentInstaller"
-                $installerFileName = "PCoIP_agent_release_installer_2.7.0.4060_standard.exe"
-                Invoke-WebRequest $source -OutFile "$dest\$installerFileName"
+				#agent installer exit code 1641 require reboot machine
+				Set-Variable EXIT_CODE_REBOOT 1641 -Option Constant
+
+                $pcoipAgentInstallerUrl = $using:pcoipAgentInstallerUrl
+                $installerFileName = [System.IO.Path]::GetFileName($pcoipAgentInstallerUrl)
+                $destFile = "C:\WindowsAzure\PCoIPAgentInstaller\" + $installerFileName
+                
+				Write-Verbose "Downloading PCoIP Agent"
+                Invoke-WebRequest $pcoipAgentInstallerUrl -OutFile $destFile
 
                 #install the agent
-                & "$dest\$installerFileName" /S
+				Write-Verbose "Installing PCoIP Agent"
+                $ret = Start-Process -FilePath $destFile -ArgumentList "/S /nopostreboot" -PassThru -Wait
+
+				# Check installer return code
+				if ($ret.ExitCode -ne 0) {
+					#exit code 1641 means requiring reboot machine after intallation is done, other non zere exit code means installation has some error
+					if ($ret.ExitCode -eq $EXIT_CODE_REBOOT) {
+						Write-Verbose "Request reboot machine after Installing pcoip agent."
+						# Setting the global:DSCMachineStatus = 1 tells DSC that a reboot is required
+						$global:DSCMachineStatus = 1
+					} else {
+						$errMsg = "Failed to install PCoIP Agent. Exit Code: " + $ret.ExitCode
+						Write-Verbose $errMsg
+						throw $errMsg
+					}
+				}
+				
+	            Write-Verbose "Finished PCoIP Agent Installation"
+            }
+        }
+
+        Script Register
+        {
+            DependsOn  = @("[Script]Install_PCoIPAgent")
+
+            GetScript  = { return 'registration'}
+            
+            TestScript = { 
+                cd "C:\Program Files (x86)\Teradici\PCoIP Agent"
+ 	            $ret = & .\pcoip-validate-license.ps1
+
+				# the powershell variable $? to indicate the last executing command status
+				return $?
+            }
+
+            SetScript  = {
+                #register code is stored at the password property of PSCredential object
+                $registrationCode = ($using:registrationCodeCredential).GetNetworkCredential().password
+                if ($registrationCode) {
+					# Insert a delay before activating license
+	                cd "C:\Program Files (x86)\Teradici\PCoIP Agent"
+
+					Write-Verbose "Activating License Code"               
+ 	                $ret = & .\pcoip-register-host.ps1 -RegistrationCode $registrationCode
+					$isExeSucc = $?
+
+					if ($isExeSucc) {
+						Write-Verbose "succeeded to activate License Code." 
```

Comment:
updated.
yes that's all we need to activate (ref: https://confluence.teradici.com/pages/viewpage.action?spaceKey=TSW&title=FNE+Licenses+for+Development)